reth/download-apply-snapshot.sh

```diff
@@ -0,0 +1,116 @@
+#!/bin/bash
+set +e
+set -u
+
+# Set alias for echoBanner if unavailable - for local testing
+[[ $(type -t echoBanner) == function ]] || alias echoBanner=echo
+
+# Translate APPLY_SNAPSHOT to uppercase; default to FALSE
+readonly APPLY_SNAPSHOT=$(echo "${APPLY_SNAPSHOT:-false}" | tr "[:lower:]" "[:upper:]")
+
+if [[ "$APPLY_SNAPSHOT" == "FALSE" ]]; then
+  echo "Automatic snapshot application disabled; to enable, set 'APPLY_SNAPSHOT=true' and restart"
+  exit 0
+fi
+
+if [[ "${RETH_DATA_DIR-x}" == x ]]; then
+  echo "RETH_DATA_DIR is undefined"
+  exit 1
+fi
+
+if [[ "${CLIENT}" != "reth" ]]; then
+  echo "Error: this script is only for reth"
+  exit 14
+fi
```

Comment:
1. IMO, `CLIENT` validation should go before `RETH_DATA_DIR` validation. Parallel validation should be added to the geth script too.
2. `SNAPSHOT_TYPE` validation should be added here only for the reth script to avoid any further processing.
3. The `# Verify that the snapshot is a datadir snapshot` step below can be removed.
4. For consistency, could you please prefix `Error: ` to all the error message in both the scripts where we've a non-zero exit?